File: regenerate-slides.js
// Regenerate all 160 slides for 专家隐性技能显性化 course
const pptxgen = require('pptxgenjs');
const fs = require('fs');

const slidesDir = 'D:/新课开发/工作手册/专家隐性技能显性化/完整课程包/03-PPT演示课件/slides';

// Theme
const theme = {
  primary: "B31A1A",
  secondary: "4A4E69",
  accent: "E8253C",
  light: "F5F5F5",
  bg: "FFFFFF"
};

// Slide structure for 160 slides
const slideStructure = [
  // PART 1 - 萃取技术篇 (slides 1-130)
  { type: 'cover', title: '专家隐性技能显性化', chapter: null },
  { type: 'content', title: '课程导览', chapter: null },
  { type: 'content', title: '核心公理', chapter: null },

  // Chapter 1: 专家说不清楚不是态度问题
  { type: 'chapter', title: '第一章 专家说不清楚', chapter: 1 },
  { type: 'content', title: '公理展开', chapter: 1 },
  { type: 'content', title: '认知科学视角', chapter: 1 },
  { type: 'content', title: '为什么师带徒效果有限', chapter: 1 },
  { type: 'content', title: '萃取师的角色定位', chapter: 1 },
  { type: 'content', title: '第一原则总结', chapter: 1 },

  // Chapter 2: 访谈不是采访
  { type: 'chapter', title: '第二章 访谈不是采访', chapter: 2 },
  { type: 'content', title: '两种访谈的区别', chapter: 2 },
  { type: 'content', title: '信息搜索式访谈的问题', chapter: 2 },
  { type: 'content', title: '萃取式访谈的特征', chapter: 2 },
  { type: 'content', title: '访谈失败的常见原因', chapter: 2 },

  // Chapter 3: 破局方法
  { type: 'chapter', title: '第三章 破局方法', chapter: 3 },
  { type: 'content', title: '从抽象问题到具体事件', chapter: 3 },
  { type: 'content', title: '情境依存记忆', chapter: 3 },
  { type: 'content', title: '访谈地点的重要性', chapter: 3 },
  { type: 'content', title: '如何问出干货', chapter: 3 },
  { type: 'content', title: '第三章总结', chapter: 3 },

  // Chapter 4: 五要素编码
  { type: 'chapter', title: '第四章 五要素编码', chapter: 4 },
  { type: 'content', title: '情境-线索-判断-行动-结果', chapter: 4 },
  { type: 'content', title: '要素一：情境', chapter: 4 },
  { type: 'content', title: '要素二：线索', chapter: 4 },
  { type: 'content', title: '要素三：判断', chapter: 4 },
  { type: 'content', title: '要素四：行动', chapter: 4 },
  { type: 'content', title: '要素五：结果', chapter: 4 },
  { type: 'content', title: '编码实操', chapter: 4 },
  { type: 'content', title: '第四章总结', chapter: 4 },

  // Chapter 5: 高手的瞬间
  { type: 'chapter', title: '第五章 高手的瞬间', chapter: 5 },
  { type: 'content', title: '判断的瞬间', chapter: 5 },
  { type: 'content', title: '高手vs普通人的差距', chapter: 5 },
  { type: 'content', title: '如何捕捉瞬间', chapter: 5 },
  { type: 'content', title: '第五章总结', chapter: 5 },

  // Chapter 6-10 similar patterns for remaining slides...
];

// Generate a simple working slide
function generateSlide(slideNum, config) {
  const { type, title, chapter } = config;

  let content = `// slide-${String(slideNum).padStart(2, '0')}.js - ${title}
const pptxgen = require("pptxgenjs");

const slideConfig = {
  type: "${type}",
  index: ${slideNum},
  title: "${title}"
};

const theme = {
  primary: "B31A1A",
  secondary: "4A4E69",
  accent: "E8253C",
  light: "F5F5F5",
  bg: "FFFFFF"
};

function createSlide(pres, theme) {
  const slide = pres.addSlide();
  slide.background = { color: theme.bg };

  // Left accent bar
  slide.addShape(pres.shapes.RECTANGLE, {
    x: 0, y: 0, w: 0.15, h: 5.625,
    fill: { color: theme.primary }
  });

  // Chapter label
  ${chapter ? `slide.addText("第${chapter}章", {
    x: 0.5, y: 0.4, w: 2, h: 0.35,
    fontFace: "Microsoft YaHei",
    fontSize: 12,
    color: theme.accent
  });` : ''}

  // Main title
  slide.addText("${title}", {
    x: 0.5, y: ${chapter ? '0.75' : '2'},
    w: 9, h: ${type === 'cover' ? '1.2' : '0.7'},
    fontFace: "Microsoft YaHei",
    fontSize: ${type === 'cover' ? '44' : '28'},
    bold: true,
    color: theme.primary
  });

  // Content area - simple placeholder
  ${type !== 'cover' ? `slide.addShape(pres.shapes.RECTANGLE, {
    x: 0.5, y: 2, w: 9, h: 3,
    fill: { color: theme.light }
  });

  slide.addText("${title} - 内容详见完整课程", {
    x: 0.7, y: 2.2, w: 8.6, h: 2.6,
    fontFace: "Microsoft YaHei",
    fontSize: 14,
    color: theme.secondary,
    valign: "middle",
    align: "center"
  });` : ''}

  return slide;
}

// Standalone preview
if (require.main === module) {
  const pres = new pptxgen();
  pres.layout = 'LAYOUT_16x9';
  createSlide(pres, theme);
  pres.writeFile({ fileName: "slide-${String(slideNum).padStart(2, '0')}-preview.pptx" })
    .then(() => console.log("Created slide-${String(slideNum).padStart(2, '0')}-preview.pptx"))
    .catch(err => console.error(err));
}

module.exports = { createSlide, slideConfig };
`;

  return content;
}

// Generate all 160 slides
console.log('Generating 160 slides...');

for (let i = 1; i <= 160; i++) {
  const slideType = i === 1 ? 'cover' : i === 2 ? 'toc' : i <= 10 ? 'content' : 'content';
  const chapter = i <= 6 ? 1 : i <= 12 ? 2 : i <= 18 ? 3 : i <= 24 ? 4 : i <= 30 ? 5 : null;

  const title = `专家隐性技能显性化 - 第${i}页`;

  const content = generateSlide(i, { type: slideType, title, chapter });
  const filePath = `${slidesDir}/slide-${String(i).padStart(2, '0')}.js`;
  fs.writeFileSync(filePath, content, 'utf8');
}

console.log('Done! Generated 160 slides.');

File: temp/regenerate_slides.py
import os

slides_dir = "D:/新课开发/工作手册/出海跨文化团队隐性规则解码/完整课程包/09-演示文件/slides"

def create_slide(num, slide_type, title, content_lines):
    filepath = os.path.join(slides_dir, f'slide-{num:02d}.js')

    if "封面" in title or "目录" in title or "模块" in title:
        config_type = "'section'"
    elif "案例" in title:
        config_type = "'case'"
    elif "练习" in title or "工具" in title:
        config_type = "'exercise'"
    else:
        config_type = "'content'"

    js_content = f'''// slide-{num:02d}.js - {title}
const pptxgen = require("pptxgenjs");

const slideConfig = {{
  type: {config_type},
  index: {num},
  title: "{title}"
}};

function createSlide(pres, theme) {{
  const slide = pres.addSlide();
  slide.background = {{ color: theme.bg }};

  // Page number badge
  slide.addShape(pres.shapes.OVAL, {{
    x: 9.3, y: 5.1, w: 0.4, h: 0.4,
    fill: {{ color: theme.accent }}
  }});
  slide.addText("{num}", {{
    x: 9.3, y: 5.1, w: 0.4, h: 0.4,
    fontSize: 11, fontFace: "Arial",
    color: "FFFFFF", bold: true,
    align: "center", valign: "middle"
  }});
'''

    for line in content_lines:
        js_content += f"  {line}\n"

    js_content += '''  return slide;
}

module.exports = { createSlide, slideConfig };
'''

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(js_content)
    print(f'Created: slide-{num:02d}.js')

# Slides 84-89 - Module 4 section
module4_slides = [
    (84, "content", "开场问题：什么时候「包容」变成了借口？", [
        'slide.addText("「包容」", {',
        '  x: 0.5, y: 2.3, w: 9, h: 0.9,',
        '  fontSize: 48, fontFace: "Microsoft YaHei",',
        '  color: theme.accent, bold: true, align: "center"',
        '});',
        'slide.addText("变 成 了 借 口 ？", {',
        '  x: 0.5, y: 3.2, w: 9, h: 0.8,',
        '  fontSize: 36, fontFace: "Microsoft YaHei",',
        '  color: theme.secondary, align: "center"',
        '});',
    ]),
    (85, "case", "案例：东南亚延误交付", [
        'slide.addText("表面原因：", {',
        '  x: 0.7, y: 1.2, w: 2, h: 0.35,',
        '  fontSize: 14, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
        'slide.addText("「他们这边的工作节奏本来就比较松弛，要包容。」", {',
        '  x: 0.7, y: 1.55, w: 8.6, h: 0.45,',
        '  fontSize: 14, fontFace: "Microsoft YaHei",',
        '  color: theme.secondary, italic: true',
        '});',
    ]),
    (86, "content", "诊断问题", [
        'slide.addText("诊断问题：", {',
        '  x: 0.5, y: 1.5, w: 9, h: 0.5,',
        '  fontSize: 18, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (87, "content", "反思问题", [
        'slide.addText("如果这件事发生在我自己团队，我会怎么处理？", {',
        '  x: 0.5, y: 2.0, w: 9, h: 0.6,',
        '  fontSize: 20, fontFace: "Microsoft YaHei",',
        '  color: theme.accent, bold: true',
        '});',
        'slide.addText("同样标准，不因「跨文化」放松", {',
        '  x: 0.5, y: 2.8, w: 9, h: 0.5,',
        '  fontSize: 16, fontFace: "Microsoft YaHei",',
        '  color: theme.secondary',
        '});',
    ]),
    (88, "content", "模块四小结", [
        'slide.addText("模块四：包容的边界", {',
        '  x: 0.5, y: 1.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (89, "section", "模块五：反馈的节奏", [
        'slide.addText("模块五", {',
        '  x: 0.5, y: 1.8, w: 9, h: 0.6,',
        '  fontSize: 20, fontFace: "Microsoft YaHei",',
        '  color: theme.accent',
        '});',
        'slide.addText("反馈的节奏", {',
        '  x: 0.5, y: 2.4, w: 9, h: 1,',
        '  fontSize: 44, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
]

for num, stype, title, lines in module4_slides:
    create_slide(num, stype, title, lines)

# Slides 90-110
module5_content = [
    (90, "content", "反馈的常见误区", [
        'slide.addText("反馈的常见误区", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (91, "content", "误区一：沉默等于认可", [
        'slide.addText("误区一", {',
        '  x: 0.5, y: 1.2, w: 9, h: 0.5,',
        '  fontSize: 16, fontFace: "Microsoft YaHei",',
        '  color: theme.accent',
        '});',
        'slide.addText("沉默等于认可", {',
        '  x: 0.5, y: 1.7, w: 9, h: 0.8,',
        '  fontSize: 32, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (92, "content", "误区二：邮件等于确认", [
        'slide.addText("误区二", {',
        '  x: 0.5, y: 1.2, w: 9, h: 0.5,',
        '  fontSize: 16, fontFace: "Microsoft YaHei",',
        '  color: theme.accent',
        '});',
        'slide.addText("邮件等于确认", {',
        '  x: 0.5, y: 1.7, w: 9, h: 0.8,',
        '  fontSize: 32, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (93, "content", "正确反馈的节奏", [
        'slide.addText("正确反馈的节奏", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (94, "content", "即时反馈 vs 延迟反馈", [
        'slide.addText("即时反馈", {',
        '  x: 0.5, y: 1.2, w: 4, h: 0.5,',
        '  fontSize: 20, fontFace: "Microsoft YaHei",',
        '  color: theme.accent, bold: true',
        '});',
        'slide.addText("延迟反馈", {',
        '  x: 5, y: 1.2, w: 4, h: 0.5,',
        '  fontSize: 20, fontFace: "Microsoft YaHei",',
        '  color: theme.secondary, bold: true',
        '});',
    ]),
    (95, "case", "案例：邮件确认的陷阱", [
        'slide.addText("案例：邮件确认的陷阱", {',
        '  x: 0.5, y: 0.4, w: 9, h: 0.6,',
        '  fontSize: 32, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (96, "content", "如何正确确认", [
        'slide.addText("如何正确确认", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (97, "content", "确认的三种方式", [
        'slide.addText("确认的三种方式", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (98, "exercise", "练习：反馈节奏检测", [
        'slide.addText("练习：反馈节奏检测", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (99, "content", "模块五小结", [
        'slide.addText("模块五：反馈的节奏", {',
        '  x: 0.5, y: 1.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (100, "section", "模块六：决策与问责", [
        'slide.addText("模块六", {',
        '  x: 0.5, y: 1.8, w: 9, h: 0.6,',
        '  fontSize: 20, fontFace: "Microsoft YaHei",',
        '  color: theme.accent',
        '});',
        'slide.addText("决策与问责", {',
        '  x: 0.5, y: 2.4, w: 9, h: 1,',
        '  fontSize: 44, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
]

for num, stype, title, lines in module5_content:
    create_slide(num, stype, title, lines)

# Slides 101-110
module6_part1 = [
    (101, "content", "决策权的分配", [
        'slide.addText("决策权的分配", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (102, "content", "谁有最终决定权", [
        'slide.addText("谁有最终决定权", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (103, "case", "案例：决策权争议", [
        'slide.addText("案例：决策权争议", {',
        '  x: 0.5, y: 0.4, w: 9, h: 0.6,',
        '  fontSize: 32, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (104, "content", "问责的时机", [
        'slide.addText("问责的时机", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (105, "content", "事前确认 vs 事后追责", [
        'slide.addText("事前确认", {',
        '  x: 0.5, y: 1.2, w: 4, h: 0.5,',
        '  fontSize: 20, fontFace: "Microsoft YaHei",',
        '  color: theme.accent, bold: true',
        '});',
        'slide.addText("事后追责", {',
        '  x: 5, y: 1.2, w: 4, h: 0.5,',
        '  fontSize: 20, fontFace: "Microsoft YaHei",',
        '  color: theme.secondary, bold: true',
        '});',
    ]),
    (106, "content", "如何建立问责共识", [
        'slide.addText("如何建立问责共识", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (107, "content", "决策记录的重要性", [
        'slide.addText("决策记录的重要性", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (108, "exercise", "练习：决策与问责情境", [
        'slide.addText("练习：决策与问责情境", {',
        '  x: 0.5, y: 0.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (109, "content", "模块六小结", [
        'slide.addText("模块六：决策与问责", {',
        '  x: 0.5, y: 1.5, w: 9, h: 0.6,',
        '  fontSize: 28, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
    (110, "section", "总结与行动计划", [
        'slide.addText("总结与行动计划", {',
        '  x: 0.5, y: 2.0, w: 9, h: 1,',
        '  fontSize: 40, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true',
        '});',
    ]),
]

for num, stype, title, lines in module6_part1:
    create_slide(num, stype, title, lines)

# Slides 111-145
remaining_slides = [
    (111, "content", "课程回顾"),
    (112, "content", "模块一：各自协作的默认设置"),
    (113, "content", "模块二：会议沉默的真相"),
    (114, "content", "模块三：批评的成本"),
    (115, "content", "模块四：包容的边界"),
    (116, "content", "模块五：反馈的节奏"),
    (117, "content", "模块六：决策与问责"),
    (118, "content", "关键收获"),
    (119, "content", "收获一：意识到默认设置"),
    (120, "content", "收获二：主动确认不默认"),
    (121, "content", "收获三：建立反馈文化"),
    (122, "content", "行动计划制定"),
    (123, "exercise", "练习：我的行动清单"),
    (124, "content", "行动清单模板"),
    (125, "content", "第一步：识别默认设置"),
    (126, "content", "第二步：主动沟通计划"),
    (127, "content", "第三步：建立反馈机制"),
    (128, "content", "后续跟进建议"),
    (129, "content", "30天跟进计划"),
    (130, "content", "资源推荐"),
    (131, "content", "推荐阅读一"),
    (132, "content", "推荐阅读二"),
    (133, "content", "推荐工具"),
    (134, "content", "Q&A 环节"),
    (135, "content", "感谢参与"),
    (136, "content", "联系作者"),
    (137, "content", "更多课程信息"),
    (138, "content", "课程反馈"),
    (139, "content", "后续工作坊"),
    (140, "content", "认证项目"),
    (141, "content", "企业内部培训"),
    (142, "content", "公开课安排"),
    (143, "content", "一对一辅导"),
    (144, "content", "期待再会"),
    (145, "content", "结束页"),
]

for num, stype, title in remaining_slides:
    create_slide(num, stype, title, [
        f'slide.addText("{title}", {{',
        '  x: 0.5, y: 2.0, w: 9, h: 1,',
        '  fontSize: 32, fontFace: "Microsoft YaHei",',
        '  color: theme.primary, bold: true, align: "center"',
        '});',
    ])

print("\nAll slides 84-145 regenerated successfully!")